test/box/gh-5518-add-granularity-option.test.lua

```diff
@@ -0,0 +1,204 @@
+-- (must be greater than or equal to 4)
+test_run:cmd('start server test with args="2" with crash_expected=True')
+-- Start server test with granularity == 7 failed (must be exponent of 2)
+test_run:cmd('start server test with args="7" with crash_expected=True')
+
+test_run:cmd('start server test with args="4"')
+test_run:cmd('switch test')
+
+s = box.schema.space.create('test')
+_ = s:create_index('test')
+
+-- Granularity determines not only alignment of objects,
+-- but also size of the objects in the pool. Thus, the greater
+-- the granularity, the greater the memory loss per one memory allocation,
+-- but tuples with different sizes are allocated from the same mempool,
+-- and we do not lose memory on the slabs, when we have highly
+-- distributed tuple sizes. This is somewhat similar to a large alloc factor
+
+-- Try to insert/delete to space, in case when UB sanitizer on,
+-- we check correct memory aligment
+for i = 1, 1000 do s:insert{i, i + 1} end
+-- Get item_size from box.slab.stats
+-- Show it and check that item_size is a multiple of granularity
+box_slab_stats = box.slab.stats()
+print("box_slab_stats for granularity = 4")
+test_run:cmd("setopt delimiter ';'")
+for _, stats in pairs(box_slab_stats) do
+    assert(type(stats) == "table")
+    for key, value in pairs(stats) do
+        if key == "item_size" then
+            print("item_size = ", value)
+            assert((value % 4) == 0)
+        end
+    end
+end;
+test_run:cmd("setopt delimiter ''");
+-- Save box.slab.info in file for further comparison
+-- file format: key "space" value
+slab_info_4 = box.slab.info()
+file = io.open("slab_info_4", "w")
+test_run:cmd("setopt delimiter ';'")
+for name, value in pairs(slab_info_4) do
+    file:write(name) file:write(" ")
+    file:write(value) file:write("\n")
+end;
+test_run:cmd("setopt delimiter ''");
+file:close()
+for i = 1, 1000 do s:delete{i} end
+s:drop()
+test_run:cmd('switch default')
+test_run:cmd('stop server test')
+
+test_run:cmd('start server test with args="64"')
+test_run:cmd('switch test')
+s = box.schema.space.create('test')
+_ = s:create_index('test')
+for i = 1, 1000 do s:insert{i, i + 1} end
+-- Get item_size from box.slab.stats
+-- Show it and check that item_size is a multiple of granularity
+box_slab_stats = box.slab.stats()
+print("box_slab_stats for granularity = 64")
+test_run:cmd("setopt delimiter ';'")
+for _, stats in pairs(box_slab_stats) do
+    assert(type(stats) == "table")
+    for key, value in pairs(stats) do
+        if key == "item_size" then
+            print("item_size = ", value)
+            assert((value % 64) == 0)
+        end
+    end
+end;
+test_run:cmd("setopt delimiter ''");
+-- Save box.slab.info in file for further comparison
+-- file format: key "space" value
+slab_info_64 = box.slab.info()
+file = io.open("slab_info_64", "w")
+test_run:cmd("setopt delimiter ';'")
+for name, value in pairs(slab_info_64) do
+    file:write(name) file:write(" ")
```

Comment:
Instead of saving values in file, you can use test_run:eval():
```
start foo
switch foo
...
switch default
test_run:eval('foo', 'return bla bla')
stop foo
```